specs/core/1_shard-data-chains.md

```diff
@@ -412,7 +592,14 @@ def is_valid_beacon_attestation(shard: Shard,
                     start_epoch + MAX_EPOCHS_PER_CROSSLINK)
     blocks = []
     for slot in range(start_epoch * SLOTS_PER_EPOCH, end_epoch * SLOTS_PER_EPOCH):
-        blocks.append(shard_blocks[slot])
+        if isinstance(shard_blocks_or_state_roots[slot], ShardBlock):
+            blocks.append(shard_blocks_or_state_roots[slot])
+        else:
+            blocks.append(ShardBlockHeader(ShardBlockCore(
+                slot=slot,
+                state_root=shard_blocks_or_state_roots[slot],
+                total_bytes=pre_state.total_bytes
+            ), ShardBlockSignatures()))
```

Comment:
It seems `blocks` should be `Sequence[ShardBlock]` type?

```python
            blocks.append(ShardBlock(
                core=ExtendedShardBlockCore(
                    slot=slot,
                    state_root=shard_blocks_or_state_roots[slot],
                    total_bytes=pre_state.total_bytes,
                ),
                signatures=ShardBlockSignatures(),
            ))
```

Reply:
Yep! Because `blocks` gets passed in to `def compute_crosslink_data_root(blocks: Sequence[ShardBlock]) -> Hash` right below.